python-saline.spec

```
@@ -20,6 +20,8 @@ BuildRequires:  python-rpm-macros
 BuildRequires:  systemd-rpm-macros
 BuildRequires:  %{python_module base}
 Requires:       %{python_module salt}
+Requires:       python3-CherryPy
+Requires:       python3-python-dateutil
```

Yup, makes sense, but let's do it the following way, just to make it more singlespec ready:
```suggestion
Requires:       %{python_module CherryPy}
Requires:       %{python_module python-dateutil}
```

---

Hi,

as far as I know `%python_module` is only intended for BuildRequires (see https://github.com/openSUSE/python-rpm-macros), not for Requires.
Otherwise a Python flavored Saline package would require dependencies for all other Python versions defined in the distributions `%pythons`. For example if you were to install `python3x-saline` on Tumbleweed, it would then require `python39-CherryPy`, `python310-CherryPy` and `python311-CherryPy` instead of only the one matching the installed Saline flavor.
This is just what I know from other Python packages, so maybe I am missing something here?

---

Seems you are closer to true than me, yes there is no need to use `%python_module` but singlespec expects `python-` prefix instead of `python3-` anyway.
Like here: https://build.opensuse.org/package/view_file/devel:languages:python/python-pip/python-pip.spec?expand=1
Lines 61, 62

---

Interesting, looking at the pip package you linked, `python-setuptools` seems to return `python310-setuptools` in `rpm -q --requires`. Not sure how/why this works, but if it does, it sounds good with me. :)

---

Updated, for the `salt` one as well.

---

Actually singlespec macro does all the magic. Basically speaking singlespec is a lua macro which rewrites the spec on the fly. And with `salt` dependency I just missed this point, forget about it and it works fine almost by accident because `salt` package is not using singlespec now and `%python_module` is expanded to `python3` only. The other flavours are not yet used.

---

Ah, I understand. Yes, with Leap/SLE it does not (yet) make a difference, but good to have it right already for compatibility with Tumbleweed. :)